python/pyspark/pandas/series.py

```diff
@@ -4475,6 +4477,146 @@ def replace(
+        Consider 2 Datasets ``s1`` and ``s2`` containing
+        highest clocked speeds of different birds.
+
+        >>> from pyspark.pandas.config import set_option, reset_option
+        >>> set_option("compute.ops_on_diff_frames", True)
+        >>> s1 = ps.Series({'falcon': 330.0, 'eagle': 160.0})
+        >>> s1
+        falcon    330.0
+        eagle     160.0
+        dtype: float64
+        >>> s2 = ps.Series({'falcon': 345.0, 'eagle': 200.0, 'duck': 30.0})
+        >>> s2
+        falcon    345.0
+        eagle     200.0
+        duck       30.0
+        dtype: float64
+
+        Now, to combine the two datasets and view the highest speeds
+        of the birds across the two datasets
+
+        >>> s1.combine(s2, max, return_type="double")
+        duck        NaN
+        eagle     200.0
+        falcon    345.0
+        dtype: float64
+
+        In the previous example, the resulting value for duck is missing,
+        because the maximum of a NaN and a float is a NaN.
+        So, in the example, we set ``fill_value=0``,
+        so the maximum value returned will be the value from some dataset.
+
+        >>> s1.combine(s2, max, fill_value=0, return_type="double")
+        duck       30.0
+        eagle     200.0
+        falcon    345.0
+        dtype: float64
+        >>> reset_option("compute.ops_on_diff_frames")
+        """
+        if not isinstance(other, Series) and not np.isscalar(other):
+            raise TypeError("unsupported type: %s" % type(other))
+
+        tmp_other_col = verify_temp_column_name(self._internal.spark_frame, "__tmp_other_col__")
+        if np.isscalar(other):
+            combined = DataFrame(
+                InternalFrame(
+                    spark_frame=self._internal.spark_frame.select(
+                        *self._internal.spark_columns
+                    ).withColumn(tmp_other_col, SF.lit(other)),
+                    index_spark_columns=self._internal.index_spark_columns,
+                )
+            )
+        elif same_anchor(self, other):
+            combined = self._psdf[self._column_label, other._column_label]
+        elif fill_value is None:
+            combined = combine_frames(self.to_frame(), other.to_frame())
+        else:
+            combined = self._combine_frame_with_fill_value(other, fill_value=fill_value)
+
+        @pandas_udf(returnType=return_type)  # type: ignore
+        def wrapped_func(x: pd.Series, y: pd.Series) -> pd.Series:
+            return x.combine(y, func)
+
+        internal = InternalFrame(
+            spark_frame=combined._internal.spark_frame.select(
+                *combined._internal.index_spark_columns,
+                wrapped_func(*combined._internal.data_spark_columns),
+                NATURAL_ORDER_COLUMN_NAME
+            ),
+            index_spark_columns=combined._internal.index_spark_columns,
+            column_labels=self._internal.column_labels if np.isscalar(other) else [None],
+            column_label_names=[None],
```

Comment:
We should keep the name when the `other` has same name, or `other` is scalar.

1. when the `other` has same name.
```python
>>> self
falcon    330.0
eagle     160.0
Name: pandas-on-Spark, dtype: float64
>>> other
falcon    345.0
eagle     200.0
duck       30.0
Name: pandas-on-Spark, dtype: float64

>>> self.to_pandas().combine(other.to_pandas(), max)
duck        NaN
eagle     200.0
falcon    345.0
Name: pandas-on-Spark, dtype: float64  # The name is kept
```

2. when the `other` is scalar.
```python
>>> self.to_pandas().combine(3, max)
falcon    330.0
eagle     160.0
Name: pandas-on-Spark, dtype: float64
```

Reply:
updated